File: solutions/028/_028.java
public class _028 {
    public static void main(String[] args) {
        int n = 1001;
        int sum = 0;
        for (int i = 1; a(i) <= n*n; i++)
            sum += a(i);
        System.out.println(sum);
    }
    
    private static int a(int n) {
        return n*(n+2)/4 + (n%4)/3 + 1;
    }
}

File: solutions/028/028.rs
fn a(n: i64) -> i64 {
    return n*(n+2)/4 + (n%4)/3 + 1;
}

fn main() {
    let n = 1001;
    let mut sum = 0;
    let mut i = 1;
    while a(i) <= n*n {
        sum += a(i);
        i += 1;
    }
    println!("{}", sum);
}
